cmd/theme/main.go

```diff
@@ -50,79 +41,84 @@ var permittedCommands = map[string]string{
 	"update":                      "Update application",
 }
 
-type CommandParser func(string, []string) (map[string]interface{}, *flag.FlagSet)
-
-var parserMapping = map[string]CommandParser{
-	"upload":    FileManipulationCommandParser,
-	"download":  FileManipulationCommandParser,
-	"remove":    FileManipulationCommandParser,
-	"replace":   FileManipulationCommandParser,
-	"watch":     WatchCommandParser,
-	"configure": ConfigurationCommandParser,
-	"bootstrap": BootstrapParser,
-	"version":   NoOpParser,
-	"update":    NoOpParser,
-}
-
-type Command func(map[string]interface{}) chan bool
-
-var commandMapping = map[string]Command{
-	"upload":    commands.UploadCommand,
-	"download":  commands.DownloadCommand,
-	"remove":    commands.RemoveCommand,
-	"replace":   commands.ReplaceCommand,
-	"watch":     commands.WatchCommand,
-	"configure": commands.ConfigureCommand,
-	"bootstrap": commands.BootstrapCommand,
-	"version":   commands.VersionCommand,
-	"update":    commands.UpdateCommand,
-}
-
-func CommandDescription() string {
-	commandDescription := make([]string, len(commandDescriptionPrefix)+len(permittedCommands))
-	pos := 0
-	for i := range commandDescriptionPrefix {
-		commandDescription[pos] = commandDescriptionPrefix[i]
-		pos++
-	}
-
-	for cmd, desc := range permittedCommands {
-		commandDescription[pos] = fmt.Sprintf("    %s:\n        %s", cmd, desc)
-		pos++
-	}
-
-	return strings.Join(commandDescription, "\n")
-}
+// CommandParser maps functions that take a command name and string args, to a prepared Args struct and set of CLI flags
+type CommandParser func(string, []string) (commands.Args)
```

Comment:
Just a thought... is it parsing commands? Or is it parsing arguments? I mean the return type is Args, so maybe it should be an ArgsParser?